libvisual-plugins/plugins/actor/gstreamer/actor_gstreamer.c

```diff
@@ -90,26 +91,41 @@ static int act_gstreamer_init (VisPluginData *plugin)
 
     gst_init (NULL, NULL);
 
-    char *launch_str = g_strdup_printf ("filesrc location=%s ! decodebin ! ffmpegcolorspace ! "
+    // Regarding the pipeline below:
+    // - element "filesrc" reads a video file from disk
+    // - element "decodebin" detects the video file's encoding
+    //   and serves raw pixel data
+    // - element "videoconvert" does colorspace conversion from given to wanted
```

Comment:
It would be more accurate to say `videoconvert` performs video format conversion. After all, it could convert RGB to BGR - the colour channel order  changes but the colourspace remains RGB.

Reply:
This was inspired by gst-inspect-1.0 output where it says "Converts video from one colorspace to another". You have  a point though, pushed.

Reply:
@hartwork, that's odd. I got the language from `videoconvert`'s online documentation [here](https://gstreamer.freedesktop.org/documentation/videoconvertscale/videoconvert.html?gi-language=c).

Reply:
@kaixiong the object hierachy slighty differs over here and also the upper limit to `width` and `height`:
```console
# gst-inspect-1.0 videoconvert
Factory Details:
  Rank                     none (0)
  Long-name                Colorspace converter
  Klass                    Filter/Converter/Video
  Description              Converts video from one colorspace to another
  Author                   GStreamer maintainers <gstreamer-devel@lists.freedesktop.org>

Plugin Details:
  Name                     videoconvert
  Description              Colorspace conversion
  Filename                 /usr/lib64/gstreamer-1.0/libgstvideoconvert.so
  Version                  1.20.5
  License                  LGPL
  Source module            gst-plugins-base
  Source release date      2022-12-19
  Binary package           Gentoo GStreamer ebuild
  Origin URL               https://www.gentoo.org

GObject
 +----GInitiallyUnowned
       +----GstObject
             +----GstElement
                   +----GstBaseTransform
                         +----GstVideoFilter
                               +----GstVideoConvert

[..]
```